Library/Homebrew/extend/os/mac/formula_cellar_checks.rb

```diff
@@ -59,6 +59,33 @@ def check_python_framework_links(lib)
     EOS
   end
 
+  def check_python_virtualenv
+    framework = formula.libexec/".Python"
+    return unless framework.symlink?
+    return unless framework.realpath.to_s.start_with?("/System")
+    <<-EOS.undent
+      virtualenv created against system Python
+      This formula created a virtualenv using system Python.
+      Please add `depends_on :python` to the formula.
```

Comment:
By "on the test bot" you mean "when building bottles", right? I'm not 100% sure if that happens if e.g. `bottle :disabled` or `bottle :unused` is set; may be worth testing.